migrations/dimensions-config/1fae088c80b6.py

```diff
@@ -0,0 +1,877 @@
+        ),
+        *check_constraints,
+        schema=ctx.schema,
+    )
+
+    # Populate group table based on the data in the exposure table.
+    _LOG.info("Populating group table")
+    exposure_table = ctx.get_table("exposure")
+    select = sa.select(
+        exposure_table.columns["instrument"],
+        exposure_table.columns["group_name"],
+    ).distinct()
+    op.execute(
+        table.insert().from_select(
+            [
+                "instrument",
+                "name",
+            ],
+            select,
+        )
+    )
+
+    # Create index on instrument
+    _LOG.info("Creating instrument index for group table")
+    op.create_index(
+        "group_fkidx_instrument",
+        "group",
+        ["instrument"],
+        schema=ctx.schema,
+    )
+
+    # Update the exposure table to reference the group table.
+    _LOG.info("Updating exposure table to reference group table")
+    with op.batch_alter_table("exposure", schema=ctx.schema) as batch_op:
+        batch_op.alter_column("group_name", new_column_name="group", nullable=False)
+        batch_op.drop_column("group_id")
+
+    # In theory we should do this create_foreign_key as part of the batch
+    # above.  However, there is some undocumented weirdness with the column
+    # rename from "group_name" to "group".  When done in the batch above, this
+    # foreign key only works if you specify the original column name instead of
+    # the final one.  This seems fragile (and is likely incompatible with
+    # Postgres, which ignores the batching). So do it in a separate batch.
+    with op.batch_alter_table("exposure", schema=ctx.schema) as batch_op:
+        batch_op.create_foreign_key(
+            constraint_name="fkey_exposure_group_instrument_name_instrument_group",
+            referent_table="group",
+            local_cols=["instrument", "group"],
+            remote_cols=["instrument", "name"],
+            referent_schema=ctx.schema,
+        )
+
+    # Create index on exposure for group fkey
+    op.create_index(
+        "exposure_fkidx_instrument_group",
+        "exposure",
+        ["instrument", "group"],
+        schema=ctx.schema,
+    )
+
+
+def _migrate_day_obs(ctx: _Context) -> None:
+    # Before doing anything else, generate the rows for the new day_obs table
+    # from the data in the exposure table.  This is prone to failure due to the
+    # need to import instrument classes.
+    _LOG.info("Generating data for day_obs table from exposure_table")
+    day_obs_rows = list(_generate_day_obs_rows(ctx))
+
+    # Create day_obs table
+    _LOG.info("Creating day_obs table")
+    timespan_columns: list[sa.Column]
+    if ctx.is_postgres:
+        # Postgres uses a non-standard range datatype for representing timespans.
+        timespan_columns = [sa.Column("timespan", INT8RANGE)]
+    else:
+        timespan_columns = [
+            sa.Column("timespan_begin", sa.BigInteger),
+            sa.Column("timespan_end", sa.BigInteger),
+        ]
```

Comment:
This (and line 237-243 below) could go into a special little module (`lsst.daf.butler_migrate.timespan`) as we may also need it for future migrations.